BLE_GattServer_FOTAService/README.md

```diff
@@ -1,36 +1,72 @@
-    ```
+### Hardware Requirements
 
-1. CLI 1 outputs a hex file to the BUILD folder, but we need a bin. Manually create it from the elf using objcopy:
+This application requires either the [DISCO_L475VG_IOT01A](https://os.mbed.com/platforms/ST-Discovery-L475E-IOT01A/) or [NRF52840_DK](https://os.mbed.com/platforms/Nordic-nRF52-DK/) platforms.
 
-    ```shell
-    arm-none-eabi-objcopy -O binary BUILD/NRF52840_DK/GCC_ARM/BLE_GattServer_FOTAService.elf BUILD/NRF52840_DK/GCC_ARM/BLE_GattServer_FOTAService.bin
-    ```
-   The file should be approx. 325 KiB in size.
+To use the Python client, the host PC must have Bluetooth capabilities, either through an inbuilt chipset or an external USB adapter.
 
-1. Create a Python virtual environment and activate it. Then, install [bleak](https://pypi.org/project/bleak/).
+### Building 
 
-   ```shell
-   cd scripts && mkdir venv && virtualenv venv && source venv/bin/activate  
-   pip install bleak==0.10.0  
-   ```
+The application can be built and flashed onto the target using Mbed CLI 2:
+
+```shell
+mbedtools deploy
+mbedtools compile -t <toolchain> -m <target> -f
+```
+
+A `bin` file is required for the demo. 
+For nRF, this must be created manually from the `elf` as Mbed Tools outputs a `hex` file by default:
+
+```shell
+arm-none-eabi-objcopy -O binary cmake_build/NRF52840_DK/develop/<toolchain>/BLE_GattServer_FOTAService.elf cmake_build/NRF52840_DK/develop/<toolchain>/BLE_GattServer_FOTAService.bin
+```
+
+Lastly, create a Python virtual environment inside the `scripts` folder and install bleak:
+
+```shell
+cd scripts && mkdir venv && virtualenv venv && source venv/bin/activate
+python -m pip install --upgrade pip && pip install bleak
+```
+
+### Demonstration
+
+1. Open a serial terminal on the host PC to receive serial prints from the _FOTA target_: 
    
-4. Run the test script. It will guide you through the process of transferring the binary.
+   ```shell 
+   mbed term -b 115200
+   ```
 
-   ```shell
+2. In a separate window, run the test script: 
+   
+   ```
    python test_fota.py
    ```
-
-5. Once you commit the update, the target will compute its SHA256 and print it to the serial. Verify the hash with sha256sum.
-
+   
+   It scans for a device named 'FOTA' and attempts to connect to it.
+   Once connected, it asks the user to enter the path to the binary.
+   Use the binary running on the target:
+   
+   ```
+   Enter the path to the binary: ../cmake_build/<target>/develop/<toolchain>/BLE_GattServer_FOTAService.bin
+   ```
+   
+3. The client initiates the transfer once the FOTA session begins and commits the update once the entire binary has been sent.
+   Subsequently, the target computes the SHA-256 of the binary and prints it to the serial.
+   Verify the `<hash>` with sha256sum:
+   
    ```shell
-   sha256sum BUILD/NRF52840_DK/GCC_ARM/BLE_GattServer_FOTAService.bin
+   echo "<hash> ../cmake_build/<target>/develop/<toolchain>/BLE_GattServer_FOTAService.bin" | sha256sum --check
```

Comment:
Why not compute the hash with `hashlib`: https://docs.python.org/3/library/hashlib.html ?